### PR TITLE
Add -X XMETHOD to enter-chroot

### PR DESCRIPTION
**Pull request by aetsher**
Add '-X' option to override the the auto-detected XMETHOD for this session/invocation.

### CODE REVIEW

File: host-bin/enter-chroot [hunk 1]
```diff
@@ -33,6 +34,7 @@ Options:
     -n NAME     Name of the chroot to enter. Default: first one found in CHROOTS
     -t TARGET   Only enter the chroot if it contains the specified TARGET.
     -u USERNAME Username (or UID) to log into. Default: 1000 (the primary user)
+    -X XMETHOD  For this session, override the the auto-detected XMETHOD
```
Comment (dnschneid) [inline host-bin/enter-chroot]: `Override the auto-detected XMETHOD for this session.`

File: host-bin/enter-chroot [hunk 2]
```diff
@@ -71,6 +73,7 @@ while getopts 'bc:k:ln:t:u:x' f; do
     n) NAME="$OPTARG";;
     t) TARGET="$OPTARG";;
     u) USERNAME="$OPTARG";;
+    X) TMPXMETHOD="$OPTARG";;
```
Comment (dnschneid) [inline host-bin/enter-chroot]: Need to add `X:` to the getopts string.

File: host-bin/enter-chroot [hunk 3]
```diff
@@ -676,6 +679,12 @@ else
     else
         # Escape out the command
         cmd="export SHELL='$CHROOTSHELL';"
+        if [ -n "$TMPXMETHOD" ]; then
+-            case "$TMPXMETHOD" in
+-                xiwi|xorg|xephyr) cmd="$cmd export XMETHOD='$TMPXMETHOD';" ;;
+-                *) echo "WARNING: XMETHOD '$TMPXMETHOD' invalid so not set." ;;
```
Comment (dnschneid) [inline host-bin/enter-chroot]: Please validate the parameter way earlier (where the rest of the parameters are validated), and also check if the desired target is actually installed (using .crouton-targets).

**Comment by dnschneid** [comment]
@DennisLfromGA added feedback (you might need to manually enable notifications for this PR)

File: host-bin/enter-chroot (hunk 1 above)
Comment (aetsher) [inline host-bin/enter-chroot]: Override the auto-detected XMETHOD for this session.

Comment (aetsher) [inline host-bin/enter-chroot]: Override the auto-detected XMETHOD for this session.

File: host-bin/enter-chroot (hunk 2 above)
Comment (aetsher) [inline host-bin/enter-chroot]: x:

File: host-bin/enter-chroot (hunk 1 above)
Comment (dnschneid) [inline host-bin/enter-chroot]: @DennisLfromGA needs to make the changes in the source, but you're welcome to keep trying in the comment box.

File: host-bin/enter-chroot (hunk 2 above)
Comment (DennisLfromGA) [inline host-bin/enter-chroot]: Done, I had it my original file but it didn't make it to the online version somehow.

File: host-bin/enter-chroot [hunk 4]
```diff
@@ -133,6 +136,13 @@ elif [ -n "$TARGET" ]; then
     fi
 fi
 
+# Check to ensure that the XMETHOD requested has been installed
+if [ -n "$TMPXMETHOD" ]; then
+    if ! grep -q "^$TMPXMETHOD$" "$CHROOTS/$NAME/.crouton-targets" 2>/dev/null; then
```
Comment (dnschneid) [inline host-bin/enter-chroot]: Use `"$CHROOT"` instead of `"$CHROOTS/$NAME"`

**Comment by dnschneid** [comment]
@drinkcat: are you ok with using -X for this?

**Comment by drinkcat** [comment]
> @drinkcat: are you ok with using -X for this?

Sure.

File: host-bin/enter-chroot (hunk 4 above)
Comment (DennisLfromGA) [inline host-bin/enter-chroot]: I can definitely do that but I wasn't certain that $CHROOT was valid until the routine at line 166 '# Mount the chroot and update our CHROOT path' executed.

Comment (dnschneid) [inline host-bin/enter-chroot]: Oops, you're right.  Disregard that comment.

Comment (DennisLfromGA) [inline host-bin/enter-chroot]: Disregarded :)

**Pull request merged**